File: src/topic_demo/src/talker.cpp
#include <ros/ros.h>
#include <topic_demo/gps.h>

int main(int argc, char** argv){
  setlocale(LC_ALL, "zh_CN.UTF-8");

  ros::init(argc, argv, "talker");
  ros::NodeHandle nh;

  topic_demo::gps msg;
  msg.x = 1.0;
  msg.y = 1.0;
  msg.status = "working";

  ros::Publisher pub = nh.advertise<topic_demo::gps>("gps_info", 1);

  ros::Rate r(1.0);

  while(ros::ok()){
    ROS_INFO("发布者: x=%f y=%f", msg.x, msg.y);
    pub.publish(msg);
  }

  return 0;
}

File: src/topic_demo_py/src/talker.py
import rospy
from topic_demo_py.msg import gps

def talker():
  rospy.init_node("pytalker")

  pub = rospy.Publisher("gps_info", gps, queue_size = 10)

  rate = rospy.Rate(1)

  while not rospy.is_shutdown():
    pub.publish(gps(1.0, 1.0, "working"))
    rospy.loginfo("sending...")
    rate.sleep()

if __name__ == "__main__":
  talker()
